app/controllers/articles_controller.rb

```diff
@@ -1,9 +1,37 @@
 class ArticlesController < ApplicationController
+    protect_from_forgery
+    before_action :check_journalist, only: [:create, :new]
+    before_action :authenticate_user!, only: [:create, :new]
     def index
         @articles = Article.all
     end
-    
+
+    def new
+        
+    end
+
     def show
         @article = Article.find(params[:id])
     end
+
+    def create
```

Comment:
Rewrite this method like this and the sad path will work with not seeing the success message. Also are you sure you want to redirect to new_article_path after successfully creating article?
```
    def create
        article = Article.new(article_params)
       
        if article.save
            redirect_to new_article_path, notice: 'Article was successfully created.'
        else
            redirect_to new_article_path, alert: 'You have to fill out all the fields'
        end
    end
```